### ISSUE TITLE
 Neo.ClientError.Security.Unauthorized after docker  restart

### ISSUE THREAD

**Issue by geertschneider** [neo4j/docker-neo4j#163]
Hi,

after the container is stoped/started again it seems impossible to login again (via Neo4j Browser).
I always get following error - although the username/password is correct.
 Neo.ClientError.Security.Unauthorized: The client is unauthorized due to authentication failure.

Am I doing something wrong ?

**Comment by LucasBoudereaux** [comment]
Same issue : 

This is my docker-compose.yml file : 

```
version: '3'
services:
  neo4j:
    container_name: neo4j-lab
    image: neo4j:latest
    environment:
      - NEO4J_dbms_memory_pagecache_size=2G
      - NEO4J_dbms_memory_heap_maxSize=4G
      - NEO4J_dbms_memory_heap_initialSize=512M
    ports:
      - 7474:7474
      - 7687:7687
    environment:
      - NEO4J_AUTH=neo4j/changeme
    volumes:
      - neo4j_data:/data
      - neo4j_conf:/conf
      - ./import:/import
volumes:
  neo4j_data:
  neo4j_conf:

**Comment by Akhu** [comment]
Same issue here :( Tried also without docker on MacOS Mojave

**Comment by jennyowen** [comment]
I found a setting password bug, and I think it may have been causing this problem. 

You would hit the bug if you start the docker container without the `--user` flag and while setting a password, eg:
```
docker run \
    --publish=7474:7474 --publish=7687:7687 \
    --volume=$HOME/neo4j/data:/data \
    --volume=$HOME/neo4j/logs:/logs \
    --env=NEO4J_AUTH=neo4j/newpassword \
    neo4j:3.5
```

Running Neo4j with the `--user` flag does NOT hit the bug. For example:
```
docker run \
    --publish=7474:7474 --publish=7687:7687 \
    --volume=$HOME/neo4j/data:/data \
    --volume=$HOME/neo4j/logs:/logs \
    --env=NEO4J_AUTH=neo4j/newpassword \
    --user="$(id -u):$(id -g)" \
    neo4j:3.5
```
(See https://neo4j.com/docs/operations-manual/current/docker/introduction/#docker-user)

I fixed the bug in pull request #167 so it should be fixed whenever we next release 3.5. 
Until then, try adding the `--user` flag as a workaround. If that doesn't work, please let me know here.

**Comment by jennyowen** [comment]
This will be in 3.4.14 and 3.5.6 as soon as our images are accepted onto dockerhub.
That's currently waiting on this PR:
https://github.com/docker-library/official-images/pull/5966

**Comment by SaltedEggIndomee** [comment]
@jennyowen How about Kubernetes??

Where should we provide the workaround to include the --user flag??

My Pod started up correctly with the output:
Changed password for user 'neo4j'.

But once I login to launch cypher-shell, the password is still the default 'neo4j'.

**Comment by jennyowen** [comment]
Hi all, our updates finally got released, so the issue is now fixed in `latest`, `3.5`, `3.4`, `3.4.14`, `3.5.6` and the `-enterprise` variants.

The work around from https://github.com/neo4j/docker-neo4j/issues/163#issuecomment-491282701 should no longer be necessary.

Let me know if you still encounter the problem *on those images*.
